src/v/iceberg/merge_append_action.cc

```diff
@@ -25,17 +25,17 @@
 namespace iceberg {
 
 namespace {
-manifest_path get_manifest_path(
-  const ss::sstring& location, const uuid_t& commit_uuid, size_t num) {
-    return manifest_path{
-      fmt::format("{}/metadata/{}-m{}.avro", location, commit_uuid, num)};
+uri get_manifest_path(
+  const uri& location, const uuid_t& commit_uuid, size_t num) {
+    return uri(
+      fmt::format("{}/metadata/{}-m{}.avro", location, commit_uuid, num));
 }
-manifest_list_path get_manifest_list_path(
-  const ss::sstring& location,
+uri get_manifest_list_path(
+  const uri& location,
   snapshot_id snap_id,
   const uuid_t& commit_uuid,
   size_t num) {
-    return manifest_path{fmt::format(
+    return uri{fmt::format(
```

Comment:
We need to update to_action_errc with invalid_uri (I guess io_failed?) Also in the filesystem catalog

Reply:
right, sorry for this

Reply:
sorry for this